src/components/screens/LockScreen.tsx

```diff
@@ -0,0 +1,155 @@
+  border-radius: ${px(24)};
+  padding: ${px(24)};
+  gap: ${px(8)};
+`;
+
+const ForgotPasswordContainer = styled.View`
+  margin-bottom: ${px(32)};
+`;
+
+const StyledInputContainer = styled.View`
+  width: 100%;
+  gap: ${px(12)};
+  margin-top: ${px(32)};
+`;
+
+const StyledTitle = styled(Text)`
+  font-size: ${fs(24)};
+  font-weight: 600;
+`;
+
+const StyledErrorText = styled(Text)`
+  color: ${THEME.colors.status.error};
+`;
+
+export const LockScreen: React.FC<LockScreenProps> = ({ navigation }) => {
+  const {
+    signIn,
+    isLoading: isSigningIn,
+    error,
+    authStatus,
+  } = useAuthenticationStore();
+  const [publicKey, setPublicKey] = useState<string | null>(null);
+  const [passwordValue, setPasswordValue] = useState("");
+  const { t } = useAppTranslation();
+
+  // Monitor auth status changes to navigate when unlocked
+  useEffect(() => {
+    if (authStatus === "AUTHENTICATED") {
+      // Add a small delay to ensure state is settled before navigation
+      const navigationTimeout = setTimeout(() => {
+        navigation.replace(ROOT_NAVIGATOR_ROUTES.MAIN_TAB_STACK);
+      }, 100);
+
+      return () => {
+        clearTimeout(navigationTimeout);
+      };
+    }
+    return undefined;
+  }, [authStatus, navigation]);
+
+  useEffect(() => {
+    const fetchActiveAccountPublicKey = async () => {
+      const retrievedPublicKey = await getActiveAccountPublicKey();
+      setPublicKey(retrievedPublicKey);
+    };
+
+    fetchActiveAccountPublicKey();
+  }, []);
+
+  const canContinue = useMemo(
+    () =>
+      passwordValue.length >= PASSWORD_MIN_LENGTH &&
+      passwordValue.length <= PASSWORD_MAX_LENGTH,
+    [passwordValue],
+  );
+
+  const handleUnlock = useCallback(() => {
+    if (!canContinue) return;
+
+    // Disable other navigation attempts while signing in
+    if (isSigningIn) return;
+
+    // Try to sign in - error handling is in the auth store
+    signIn({ password: passwordValue });
+    // Navigation will happen automatically through the authStatus effect
+  }, [canContinue, passwordValue, signIn, isSigningIn]);
+
+  const handlePasswordChange = useCallback((value: string) => {
+    setPasswordValue(value);
+  }, []);
+
+  return (
+    <BaseLayout useSafeArea useKeyboardAvoidingView>
+      <Container>
+        <StyledIconContainer>
+          <FreighterLogo width={px(48)} height={px(48)} />
+        </StyledIconContainer>
+        <StyledFormContainer>
+          <Avatar size="lg" publicAddress={publicKey ?? ""} />
+          <StyledTitle>{t("lockScreen.title")}</StyledTitle>
+          <Text secondary>{t("lockScreen.description")}</Text>
+          <StyledInputContainer>
+            <Input
+              isPassword
+              placeholder={t("lockScreen.passwordInputPlaceholder")}
+              fieldSize="lg"
+              value={passwordValue}
+              onChangeText={handlePasswordChange}
+            />
+            {error && <StyledErrorText>{error}</StyledErrorText>}
```

Comment:
```suggestion
            {error && <Text color={THEME.colors.status.error}>{error}</Text>}
```

I think we could probably also create an `error` Text prop in a separate PR so we could use it like `<Text error> </Text>`.  :)

Reply:
Actually, should we use `Input`'s built-in `error` prop to render the error here?